transactions.go

```diff
@@ -278,6 +278,24 @@ func (mrr *MapReadResult) Counter(key Key) (val int32, err error) {
 	return 0, fmt.Errorf("counter entry with key '%s' not found", key)
 }
 
+// MapEntry represents the key and type of a map entry (embedded CRDT).
+type MapEntry struct {
+	Key      []byte
+	CrdtType CRDTType
+}
+
+// ListMapEntries gives access to the keys and types of map entries (embedded CRDTs).
+func (mrr *MapReadResult) ListMapEntries() (val []MapEntry, err error) {
+	keyList := make([]MapEntry, len(mrr.mapResp.Entries))
+	for i, me := range mrr.mapResp.Entries {
+		keyList[i] = MapEntry{
+			Key:      me.Key.Key,
+			CrdtType: *me.Key.Type,
+		}
+	}
+	return keyList, nil
```

Comment:
As the error is always `nil`, I would change the function to only return the `[]MapEntry`.

Reply:
Addressed